src/diffusers/models/attention_processor.py

```diff
@@ -1699,6 +1701,153 @@ def __call__(
         return hidden_states
 
 
+class LuminaAttnProcessor2_0(nn.Module):
```

Comment:
about the attention processor:
1. it should not contain weights
2. it should be able to do either self-attention (when `encoder_hidden_states =None`) or cross-attention (when `encoder_hidden_states is not None` but not both at the same forward path 

It won't be easy to make it works with lumina, so let us know if you would like any help with it

Reply:
on high-level I think you need to break your `Attention` into 2: one for self-attention and another one for cross-attention, both will use`LuminaAttnProcessor2_0`

Reply:
okay, i try to make some changes on it.